core/lib/spree/core/importer/order.rb

```diff
@@ -97,21 +97,66 @@ def self.create_shipments_from_params(shipments_hash, order)
           end
         end
 
-        def self.create_line_items_from_params(line_items_hash, order)
-          return {} unless line_items_hash
-          line_items_hash.each_key do |k|
-            begin
-              extra_params = line_items_hash[k].except(:variant_id, :quantity, :sku)
-              line_item = ensure_variant_id_from_params(line_items_hash[k])
-              line_item = order.contents.add(Spree::Variant.find(line_item[:variant_id]), line_item[:quantity])
-              # Raise any errors with saving to prevent import succeeding with line items failing silently.
-              if extra_params.present?
-                line_item.update_attributes!(extra_params)
-              else
-                line_item.save!
+        def self.create_line_items_from_params(line_items, order)
+          return {} unless line_items
+          case line_items
+          when Hash
+            ActiveSupport::Deprecation.warn(<<-EOS, caller)
+              Passing a hash is now deprecated, it is recommended that you pass it as an array instead.
+
+              New Syntax:
+
+              {
+                "order": {
+                  "line_items": [
+                    { "variant_id": 123, "quantity": 1 },
+                    { "variant_id": 456, "quantity": 1 }
+                  ]
+                }
+              }
+
+              Old Syntax:
+
+              {
+                "order": {
+                  "line_items": {
+                    "1": { "variant_id": 123, "quantity": 1 },
+                    "2": { "variant_id": 123, "quantity": 1 }
+                  }
+                }
+              }
+            EOS
+
+            line_items.each_key do |k|
+              begin
+                extra_params = line_items[k].except(:variant_id, :quantity, :sku)
+                line_item = ensure_variant_id_from_params(line_items[k])
+                line_item = order.contents.add(Spree::Variant.find(line_item[:variant_id]), line_item[:quantity])
+                # Raise any errors with saving to prevent import succeeding with line items failing silently.
+                if extra_params.present?
+                  line_item.update_attributes!(extra_params)
+                else
+                  line_item.save!
+                end
+              rescue Exception => e
```

Comment:
@BenMorganIO You really should never rescue Exception. There are some cases, but this isn't one of them. Especially when you're doing a lot of stuff in this method and so much can go wrong.